File: ismgl-api/models/EtudiantModel.php
<?php
require_once __DIR__ . '/../core/Model.php';

class EtudiantModel extends Model {
    protected $table = 'etudiants';
    protected $primaryKey = 'id_etudiant';
    
    public function createEtudiant($userData, $etudiantData) {
        $this->db->beginTransaction();
        
        try {
            // Créer l'utilisateur
            $userModel = new UserModel();
            $userId = $userModel->createUser($userData);
            
            // Créer l'étudiant
            $etudiantId = $this->create([
                'id_user' => $userId,
                'numero_etudiant' => $etudiantData['numero_etudiant'],
                'date_naissance' => $etudiantData['date_naissance'],
                'lieu_naissance' => $etudiantData['lieu_naissance'] ?? null,
                'sexe' => $etudiantData['sexe'],
                'nationalite' => $etudiantData['nationalite'] ?? 'Congolaise',
                'adresse' => $etudiantData['adresse'] ?? null,
                'ville' => $etudiantData['ville'] ?? null,
                'province' => $etudiantData['province'] ?? null,
                'nom_pere' => $etudiantData['nom_pere'] ?? null,
                'nom_mere' => $etudiantData['nom_mere'] ?? null,
                'telephone_urgence' => $etudiantData['telephone_urgence'] ?? null,
                'groupe_sanguin' => $etudiantData['groupe_sanguin'] ?? null,
                'photo_identite' => $etudiantData['photo_identite'] ?? null,
                'date_premiere_inscription' => date('Y-m-d')
            ]);
            
            $this->db->commit();
            return $etudiantId;
            
        } catch (Exception $e) {
            $this->db->rollback();
            throw $e;
        }
    }
    
    public function getEtudiantComplet($id) {
        return $this->db->fetchOne(
            "SELECT e.*, u.matricule, u.nom, u.prenom, u.email, u.telephone, 
                    u.photo_profil, u.est_actif
             FROM etudiants e
             JOIN users u ON e.id_user = u.id_user
             WHERE e.id_etudiant = ?",
            [$id]
        );
    }
    
    public function getEtudiantByUser($userId) {
        return $this->db->fetchOne(
            "SELECT e.*, u.matricule, u.nom, u.prenom, u.email, u.telephone
             FROM etudiants e
             JOIN users u ON e.id_user = u.id_user
             WHERE e.id_user = ?",
            [$userId]
        );
    }
    
    public function getEtudiantByNumero($numero) {
        return $this->db->fetchOne(
            "SELECT e.*, u.nom, u.prenom, u.email
             FROM etudiants e
             JOIN users u ON e.id_user = u.id_user
             WHERE e.numero_etudiant = ?",
            [$numero]
        );
    }
    
    public function getAllEtudiants($page, $pageSize, $filters = []) {
        $offset = ($page - 1) * $pageSize;
        $params = [];
        
        $sql = "SELECT e.id_etudiant, e.numero_etudiant, e.date_naissance, e.sexe, 
                       e.statut, e.date_premiere_inscription,
                       u.matricule, u.nom, u.prenom, u.email, u.telephone, u.est_actif
                FROM etudiants e
                JOIN users u ON e.id_user = u.id_user
                WHERE 1=1";
        
        if (!empty($filters['statut'])) {
            $sql .= " AND e.statut = ?";
            $params[] = $filters['statut'];
        }
        
        if (!empty($filters['sexe'])) {
            $sql .= " AND e.sexe = ?";
            $params[] = $filters['sexe'];
        }
        
        if (!empty($filters['search'])) {
            $sql .= " AND (u.nom LIKE ? OR u.prenom LIKE ? OR e.numero_etudiant LIKE ? OR u.email LIKE ?)";
            $search = "%{$filters['search']}%";
            $params = array_merge($params, [$search, $search, $search, $search]);
        }
        
        $sql .= " ORDER BY e.date_creation DESC LIMIT ? OFFSET ?";
        $params[] = $pageSize;
        $params[] = $offset;
        
        return $this->db->fetchAll($sql, $params);
    }
    
    public function getTotalEtudiants($filters = []) {
        $params = [];
        $sql = "SELECT COUNT(*) FROM etudiants e JOIN users u ON e.id_user = u.id_user WHERE 1=1";
        
        if (!empty($filters['statut'])) {
            $sql .= " AND e.statut = ?";
            $params[] = $filters['statut'];
        }
        
        if (!empty($filters['sexe'])) {
            $sql .= " AND e.sexe = ?";
            $params[] = $filters['sexe'];
        }
        
        if (!empty($filters['search'])) {
            $sql .= " AND (u.nom LIKE ? OR u.prenom LIKE ? OR e.numero_etudiant LIKE ?)";
            $search = "%{$filters['search']}%";
            $params = array_merge($params, [$search, $search, $search]);
        }
        
        return $this->db->fetchColumn($sql, $params);
    }
    
    public function updateStatut($id, $statut) {
        return $this->update($id, ['statut' => $statut]);
    }
    
    public function generateNumeroEtudiant() {
        $annee = date('Y');
        $count = $this->db->fetchColumn(
            "SELECT COUNT(*) + 1 FROM etudiants WHERE YEAR(date_creation) = ?",
            [$annee]
        );
        
        return sprintf("ETU%s%05d", $annee, $count);
    }
}

File: ismgl/lib/data/models/etudiant_model.dart
class EtudiantModel {
  final int idEtudiant;
  final int? idUser;
  final String numeroEtudiant;
  final String matricule;
  final String nom;
  final String prenom;
  final String email;
  final String? telephone;
  final String dateNaissance;
  final String? lieuNaissance;
  final String sexe;
  final String? nationalite;
  final String? adresse;
  final String? ville;
  final String? province;
  final String? nomPere;
  final String? nomMere;
  final String? telephoneUrgence;
  final String? groupeSanguin;
  final String? photoIdentite;
  final String statut;
  final bool estActif;
  final String? datePremierInscription;
  final String? dateCreation;
  final String? dateModification;

  EtudiantModel({
    required this.idEtudiant,
    this.idUser,
    required this.numeroEtudiant,
    required this.matricule,
    required this.nom,
    required this.prenom,
    required this.email,
    this.telephone,
    required this.dateNaissance,
    this.lieuNaissance,
    required this.sexe,
    this.nationalite,
    this.adresse,
    this.ville,
    this.province,
    this.nomPere,
    this.nomMere,
    this.telephoneUrgence,
    this.groupeSanguin,
    this.photoIdentite,
    this.statut = 'Actif',
    this.estActif = true,
    this.datePremierInscription,
    this.dateCreation,
    this.dateModification,
  });

  factory EtudiantModel.fromJson(Map<String, dynamic> json) {
    int asInt(dynamic v) => v is int ? v : int.tryParse(v?.toString() ?? '0') ?? 0;
    String asString(dynamic v) => v?.toString() ?? '';
    return EtudiantModel(
      idEtudiant:            asInt(json['id_etudiant']),
      idUser:                json['id_user'] == null ? null : asInt(json['id_user']),
      numeroEtudiant:        asString(json['numero_etudiant']),
      matricule:             asString(json['matricule']).isNotEmpty
                                 ? asString(json['matricule'])
                                 : asString(json['numero_etudiant']),
      nom:                   asString(json['nom']),
      prenom:                asString(json['prenom']),
      email:                 asString(json['email']),
      telephone:             json['telephone']               as String?,
      dateNaissance:         asString(json['date_naissance']),
      lieuNaissance:         json['lieu_naissance']          as String?,
      sexe:                  asString(json['sexe']),
      nationalite:           json['nationalite']             as String?,
      adresse:               json['adresse']                 as String?,
      ville:                 json['ville']                   as String?,
      province:              json['province']                as String?,
      nomPere:               json['nom_pere']                as String?,
      nomMere:               json['nom_mere']                as String?,
      telephoneUrgence:      json['telephone_urgence']       as String?,
      groupeSanguin:         json['groupe_sanguin']          as String?,
      photoIdentite:         json['photo_identite']          as String?,
      statut:                json['statut']                  as String? ?? 'Actif',
      estActif:              json['est_actif'] == true || json['est_actif'] == 1,
      datePremierInscription: json['date_premiere_inscription'] as String?,
      dateCreation:          json['date_creation']           as String?,
      dateModification:      json['date_modification']       as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id_etudiant':              idEtudiant,
        'id_user':                  idUser,
        'numero_etudiant':          numeroEtudiant,
        'nom':                      nom,
        'prenom':                   prenom,
        'email':                    email,
        'telephone':                telephone,
        'date_naissance':           dateNaissance,
        'lieu_naissance':           lieuNaissance,
        'sexe':                     sexe,
        'nationalite':              nationalite,
        'adresse':                  adresse,
        'ville':                    ville,
        'province':                 province,
        'nom_pere':                 nomPere,
        'nom_mere':                 nomMere,
        'telephone_urgence':        telephoneUrgence,
        'groupe_sanguin':           groupeSanguin,
        'statut':                   statut,
        'est_actif':                estActif,
        'date_premiere_inscription': datePremierInscription,
      };

  String get fullName => '$prenom $nom';
  bool get isActive => statut == 'Actif';
}